tools/mpremote/mpremote/commands.py

```diff
@@ -334,6 +334,50 @@ def do_filesystem_recursive_rm(state, path, args):
             print(f"removed: '{path}'")
 
 
+def human_size(size, decimals=1):
+    for unit in ['B', 'K', 'M', 'G', 'T']:
+        if size < 1024.0 or unit == 'T':
+            break
+        size /= 1024.0
+    return f"{size:.{decimals}f}{unit}" if unit != 'B' else f"{int(size)} {unit}"
+
+
+def do_filesystem_tree(state, path, args):
+    """Print a tree of the device's filesystem starting at path."""
+    connectors = ("├── ", "└── ")
+
+    def _tree_recursive(path, prefix=""):
+        entries = state.transport.fs_listdir(path)
+        entries.sort(key=lambda e: e.name)
+        for i, entry in enumerate(entries):
+            connector = connectors[1] if i == len(entries) - 1 else connectors[0]
+            is_dir = entry.st_mode & 0x4000  # Directory
+            size_str = ""
+            # most micropyhon filesystems don't support st_size on folders - reduce clutter
+            if args.size:
+                if entry.st_size > 0 or not is_dir:
```

Comment:
You could move this if check outside, to make it a bit simpler:
```py
if entry.st_size > 0 or not is_dir:
    if args.size:
        ...
    elif args.human:
        ...
```

Reply:
The B for Bytes is actually supressed later so ot is not in the output
Indeed simpler not to add it in the first place